R/deployAPI.R

```diff
@@ -0,0 +1,31 @@
+#' Deploy a Plumber API
+#'
+#' Deploys an application consisting of a series of plumber API routes.
+#' Only one of api or entry may be passed.
+#'
+#' @param api Path to the API project directory. Must contain either `entrypoint.R` or `plumber.R`
+#' @param ... Additional arguments to \code{\link{deployApp}}.
+#'
+#' @details Deploy a plumber API definition by either supplying a directory containing `plumber.R` (an API definition)
+#'     or `entrypoint.R` that returns a `plumb` object created by `plumber::plumb()`. See the plumber
+#'     documentation for more information.
+#'
+#' @export
+deployAPI <- function(api,
+                      ...) {
+  if (!requireNamespace("plumber") ||
+      packageVersion("plumber") < "0.3.3") {
+    stop("Version 0.3.3 or later of the plumber package is required to ",
+         "deploy plumber APIs.")
+  }
+  if (!file.exists(api)) {
+    stop("The api at '", api, "' does not exist.")
+  }
+  if (!dir.exists(api)) {
```

Comment:
I think `dir.exists(api)` needs to be `utils::file_test("-d", api)`, for compatibility with older R versions